pkgs/by-name/fi/filesender/separate_mutable_paths.patch

```diff
@@ -0,0 +1,122 @@
+             include_once($pass_config_file);
+@@ -153,7 +153,7 @@ class Config
+                 throw new ConfigBadParameterException('virtualhost');
+             }
+             
+-            $config_file = FILESENDER_BASE.'/config/'.$virtualhost.'/config.php';
++            $config_file = FILESENDER_CONFIG_DIR.$virtualhost.'/config.php';
+             if (!file_exists($config_file)) {
+                 throw new ConfigFileMissingException($config_file);
+             } // Should exist even if empty
+@@ -175,7 +175,7 @@ class Config
+                         }
+                         foreach ($regex_and_configs as $regex => $extra_config_name) {
+                                 if (preg_match('`'.$regex.'`', $auth_attrs[$attr])) {
+-                                        $extra_config_file = FILESENDER_BASE.'/config/config-' . $extra_config_name . '.php';
++                                        $extra_config_file = FILESENDER_CONFIG_DIR.'/config-' . $extra_config_name . '.php';
+                                         if (file_exists($extra_config_file)) {
+                                                 $config = array();
+                                                 include_once($extra_config_file);
+@@ -204,7 +204,7 @@ class Config
+         // Load config overrides if any
+         $overrides_cfg = self::get('config_overrides');
+         if ($overrides_cfg) {
+-            $overrides_file = FILESENDER_BASE.'/config/'.($virtualhost ? $virtualhost.'/' : '').'config_overrides.json';
++            $overrides_file = FILESENDER_CONFIG_DIR.($virtualhost ? $virtualhost.'/' : '').'config_overrides.json';
+             
+             $overrides = file_exists($overrides_file) ? json_decode(trim(file_get_contents($overrides_file))) : new StdClass();
+             
+@@ -431,7 +431,7 @@ class Config
+     public static function getVirtualhosts()
+     {
+         $virtualhosts = array();
+-        foreach (scandir(FILESENDER_BASE.'/config') as $item) {
++        foreach (scandir(FILESENDER_CONFIG_DIR) as $item) {
+             if (!preg_match('`^(.+)\.conf\.php$`', $item, $match)) {
+                 continue;
+             }
+diff --git a/config/csrf-protector-config.php b/config/csrf-protector-config.php
+index 83759ca4..ea4a3173 100755
+--- a/config/csrf-protector-config.php
++++ b/config/csrf-protector-config.php
+@@ -40,7 +40,7 @@ return array(
+ 	 // The following should be set correctly from your config.php
+ 	 // information
+ 	"jsUrl" => $config['site_url'] . "/js/csrfprotector.js",
+-	"logDirectory" => FILESENDER_BASE.'/log/',
++	"logDirectory" => FILESENDER_LOG_DIR,
+ 
+         // I found that leaving this with the _ as default
+         // caused the implicit token to be stripped
+diff --git a/includes/ConfigDefaults.php b/includes/ConfigDefaults.php
+index 733550e7..8d99b5f0 100644
+--- a/includes/ConfigDefaults.php
++++ b/includes/ConfigDefaults.php
+@@ -224,7 +224,7 @@ $default = array(
+     'log_facilities' => array(
```

Comment:
Why using the old array notation ?

Reply:
This is source of patch, so I minimized amount of changes to upstream code.

Reply:
Fair enough, I was reviewing this on the phone and didn't catch that.

Reply:
(patch just changes default path for logs from fileconfig installation to env var)